homeassistant/components/feedreader.py

```diff
@@ -60,22 +61,46 @@ def _update(self):
             elif len(self._feed.entries) > 0:
                 _LOGGER.debug('Entries available in feed %s', self._url)
                 self._publish_new_entries()
-                self._last_entry_timestamp = \
-                    self._feed.entries[0].published_parsed
             else:
                 self._log_no_entries()
+        _LOGGER.info('Fetch from feed "%s" completed', self._url)
+
+    def _update_and_fire_entry(self, entry):
+        """Update last_entry_timestamp and fire entry."""
+        # We are lucky, `published_parsed` data available,
+        # let's make use of it to publish only new available
+        # entries since the last run
+        if 'published_parsed' in entry:
```

Comment:
Which brings me to: did you test this piece of the code / should you maybe write tests?